### PR TITLE
feat(events-view): Add event type filter to the Timeline tab

### PR DESCRIPTION
**Pull request by bdauria**
## Description

This adds the possibility to filter the timeline events by their type:

filter-timeline-events

## Type of change

- [x] New functionality (non-breaking change)

## Target serie

- [x] 20.04.x
- [x] 20.10.x (master)

### CODE REVIEW

**Comment by Thebarda** [comment]
@bdauria I have two remarks:

- Sometimes filter is larger that usual (see the video below). On my side, it seems to happen only when Service Memory is selected.
event timeline filter

- I have this error when I open timeline
warning-timeline

**Review by Thebarda — APPROVED** [review]

**Comment by jdelpierre** [comment]
I haven't looked at the code yet for the moment, but, in the screenshot, you don't show a little message to indicate "no result founded" ?

**Comment by bdauria** [comment]
> I haven't looked at the code yet for the moment, but, in the screenshot, you don't show a little message to indicate "no result founded" ?

fixed:

Screenshot from 2020-08-25 16-05-41

**Review by Thebarda — APPROVED** [review]

File: www/front_src/src/Resources/Details/tabs/Timeline/index.tsx [hunk 1]
```diff
@@ -49,117 +41,139 @@ const useStyles = makeStyles((theme) => ({
...
-    }).then(({ result, meta }) => {
-      setTimeline(timeline.concat(result));
+      parameters: {
+        page,
+        limit,
+        search: getSearch(),
+      },
+    }).then((retrievedListing) => {
+      const { meta } = retrievedListing;
+      setTotal(meta.total);
```
Comment (jdelpierre) [inline www/front_src/src/Resources/Details/tabs/Timeline/index.tsx:103]: is it possible that "meta.total" does not exist here? undefined ?

**Review by jdelpierre — APPROVED** [review]

**Pull request merged**